packages/lesswrong/lib/collections/comments/schema.ts

```diff
@@ -875,7 +875,42 @@ const schema: SchemaType<"Comments"> = {
     canRead: ['guests'],
     canUpdate: ['sunshineRegiment', 'admins'],
     canCreate: ['members', 'sunshineRegiment', 'admins'],
-  } 
+  },
+
+  doppelComments: {
+    // Resolved by doppelComments resolver
+    type: Array,
+    optional: true,
+    canRead: ['guests'],
+    canCreate: ['admins'],
+    canUpdate: ['admins'],
+    hidden: true,
+  },
+
+  'doppelComments.$': {
+    type: String,
+  },
+
+  ownDoppelCommentVote: resolverOnlyField({
+    type: 'DoppelCommentVote',
+    graphQLtype: 'DoppelCommentVote',
+    canRead: ['guests'],
+    resolver: async (comment: DbComment, args: void, context: ResolverContext) => {
+      const { DoppelCommentVotes } = context;
+      return await DoppelCommentVotes.findOne({commentId: comment._id, userId: context.currentUser?._id});
+    },
+  }),
+
+  doppelCommentVotes: resolverOnlyField({
+    type: 'DoppelCommentVote[]',
+    graphQLtype: '[DoppelCommentVote]',
+    canRead: ['guests'],
```

Comment:
Are other people's votes intended to be readable to all users?

Reply:
Hm, I guess the actually access control for this data would probably need to be handled by a `checkAccess` on `DoppelCommentVotes`, combined with the `accessFilterMultiple` (inside of the `resolver`; if we take the `sqlResolver` path all of that data gets run through an `accessFilterMultiple` by default if it's collection-shaped inside of `initGraphQL.ts`, L253-283)